src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Unix/PkcsPalOpenSsl.Encrypt.cs

```diff
@@ -30,6 +30,20 @@ internal sealed partial class PkcsPalOpenSsl : PkcsPal
                 throw new PlatformNotSupportedException(SR.Cryptography_Cms_AddOriginatorCertsPlatformNotSupported);
             }
 
+            if (contentEncryptionAlgorithm.KeyLength != 0 && contentEncryptionAlgorithm.KeyLength != KeyLengths.DefaultKeyLengthForRc2AndRc4)
+            {
+                switch (contentEncryptionAlgorithm.Oid.Value)
+                {
+                    case Oids.Rc2:
+                        throw new PlatformNotSupportedException(SR.Cryptography_Cms_Rc2VariableKeyLengthPlatformNotSupported);
+                    case Oids.Rc4:
+                        throw new PlatformNotSupportedException(SR.Cryptography_Cms_Rc4VariableKeyLengthPlatformNotSupported);
+                    default:
+                        // RC2 and RC4 are the only ciphers for which we support a variable key length that we need to worry about.
```

Comment:
Made a test on this, and the desktop behavior is only set the key length for RC2 and RC4. If an invalid key length is set for AES it will just encrypt using the key length specified by the OID. If it's the case for RC2 and RC4, an exception will happen on encrypt saying that the parameters are incorrect.